source/documentation/programming-languages/java.html.md.erb

```diff
@@ -0,0 +1,43 @@
+---
+title: Java styleguide
+---
+
+# <%= current_page.data.title %>
+
+The purpose of this style guide is to provide some conventions for working on Java code within GDS. The old [Sun/Oracle Java style guide](https://www.oracle.com/technetwork/java/index-135089.html) is a useful start.
+
+Generally [IntelliJ IDEA](https://www.jetbrains.com/idea/) is used within GDS and consistency of its use helps when pairing and mob programming.
+
+We favour consistency across our code, so when considering using a new or different method or paradigm to what exists already, ensure that you have agreement of your team.
+
+## Code formatting
+
+Variable and field names should match the class they are instantiating, or have a descriptive name in the context of their use.
+
+Use the default code indenting in IntelliJ.
+
+Ensure line feeds are present at the ends of files.  This can be enabled as an option in IntelliJ: `Settings->Editor->Other->Ensure line feed at file end on save.`
+
+## Dependency injection (DI)
+
+Some programmes use [Guice](https://github.com/google/guice) as a dependency injection framework, however, consider whether use of DI is appropriate for your project.
```

Comment:
this could be expanded in the future - how might one go about evaluating whether DI is appropriate?